### PR TITLE
Move func VerifySig() to its own package => crypto.VerifySig()

### PR DESCRIPTION
**Pull request by mk-livepeer**
* Move the `VerifySig()` helper to its own `crypto` package
* Remove the DefaultSigVerifier struct from the pm package - the pm package doesn't need to be exposed to the details of how sig verification is implemented.

blocks #1325

### CODE REVIEW

File: pm/validator.go
Comment (yondonfu) [inline pm/validator.go]: I think we should keep the `SigVerifier` interface in the `pm` package so that the `pm` package defines the interface it requires for signature verification as opposed to relying on an interface described by a different package. The implementation provided by the `crypto` package can just adhere to this interface.

**Review by yondonfu — CHANGES_REQUESTED** [review]

File: pm/validator.go
Comment (mkrufky) [reply pm/validator.go]: done in c813aa883b728eafcb616f266566e29505c8d832

**Review by yondonfu — COMMENTED** [review]
Changes look good - let's squash the commits into one.

**Comment by mk-livepeer** [comment]
@yondonfu please approve so that I can use the GitHub "squash and merge" button

**Comment by yondonfu** [comment]
My understanding is that the GH "squash and merge" option enforces a particular commit message structure for multi-commit PRs that includes the titles for each of the commits. The team has been following a convention of rebasing to squash fixup commits which gives us better control over the descriptions in the commit history prior to approval + merge so let's stick with that process for this PR.

**Comment by mk-livepeer** [comment]
GitHub's "squash and merge" starts with a suggested message that I always override.

If you allow me to squash first then you might miss the possibility that I snuck a bug or something else into the pr before merge.

If you prefer that I do this in the insecure way, causing an extra step in the flow, then say so.  Otherwise, please approve - I never push pre-filled commit messages

**Comment by mk-livepeer** [comment]
I squashed and pushed.  This is insecure, like I said - you should re-review the code to make sure it's the same change that you reviewed previously.

**Comment by j0sh** [comment]
> If you allow me to squash first then you might miss the possibility that I snuck a bug or something else into the pr before merge.

FWIW, this is still possible regardless. If someone approves, then the committer can force-push and merge arbitrary changes, regardless of which Github button is used. At least this way we can double check the supposedly-final commit structure before giving final approval.

**Comment by ericxtang** [comment]
Thanks for following the current commit procedure for the sake of making progress @mk-livepeer.  I think we should all be open to change our process if we can agree on a better one.  If you want to bring this up in the next eng retro, we can talk about it as a team.

**Review by yondonfu — APPROVED** [review]
LGTM

**Pull request merged**
